### PR TITLE
Add @aws-private hostnameOverride

### PR DESCRIPTION
**Pull request by jacksontj**
Potential fix for #7172

### CODE REVIEW

File: upup/pkg/fi/nodeup/command.go [hunk 1]
```diff
@@ -462,6 +462,25 @@ func evaluateHostnameOverride(hostnameOverride string) (string, error) {
 		return domain, nil
 	}
 
+	if k == "@aws-private" {
```
Comment (jacksontj) [inline upup/pkg/fi/nodeup/command.go]: If we go with this approach, I'd also like to swap this section to also switch on the `k` -- not sure why its a lot of if blocks now :/

**Comment by jacksontj** [comment]
The failures now seem to be related to bazel; but I don't see docs anywhere on what I should be updating to make bazel happy with the new imports :/

**Comment by justinsb** [comment]
I do think #7185 is actually the one we go with - thanks for looking at this option too though!

**Pull request closed without merging**